packages/@cra-express/universal-loader/src/universal.js

```diff
@@ -5,9 +5,19 @@ const stringRenderer = require('./renderer/string-renderer').default;
 
 const craServiceName = process.env.CRA_SERVICE_NAME || 'localhost';
 const craClientPort = process.env.CRA_CLIENT_PORT || 3000;
+const defaultHtmlFilename = 'index.html';
+
+function resolveHtmlFilenameByRequest(req, options) {
+  if(!options.resolveHtmlFilenameByRequest) {
+    return defaultHtmlFilename;
+  }
+
+  return options.resolveHtmlFilenameByRequest(req) || defaultHtmlFilename;
```

Comment:
`options` is not passed?

Reply:
Not because when you call universalMiddleware you already have that data, so I think you do not need it anymore.

what do you think?

Reply:
then we don't need to pass the options here https://github.com/antonybudianto/cra-universal/pull/148/files/55157a39546c4f9f584051dc2f798e170bfd79d1#diff-81014622682dc5b735995190fdad55faR42

Reply:
Yes, because

The function "resolveHtmlFilenameByRequest" verifies that there is a function of the same name in the universalMiddleware. If there is one then it executes it with the request.

    // local function in universalMiddleware
    function resolveHtmlFilenameByRequest(req, options) {
        var defaultHtml = 'index.html';
        if(!options.resolveHtmlFilenameByRequest) {
            return defaultHtml;
        }

        // If the function exists in the options then the request passes
        return options.resolveHtmlFilenameByRequest(req) || defaultHtml;
    }

    .
    .
    .
    // Call local "resolveHtmlFilenameByRequest"
    const filePath = path.resolve(clientBuildPath, resolveHtmlFilenameByRequest(req, options));

Reply:
it should be
```
const filePath = path.resolve(clientBuildPath, resolveHtmlFilenameByRequest(req));
```
?

Reply:
Not because the options have the "resolveHtmlFilenameByRequest"

    function resolveHtmlFilenameByRequest(req, options) {
        // >>>>>>>>>>>>> here exists validate <<<<<<<<<<<<<<
        if(!options.resolveHtmlFilenameByRequest) {
            return defaultHtml;
        }

        // >>>>>>>>>>>>> here execute <<<<<<<<<<<<<<
        return options.resolveHtmlFilenameByRequest(req) || defaultHtml;
    }

Reply:
oh sorry missed that, lgtm then 👍